pkg/controller/utils/utils.go

```diff
@@ -615,3 +615,27 @@ func GetElasticsearch(ctx context.Context, c client.Client) (*esv1.Elasticsearch
 	}
 	return &es, nil
 }
+
+func IsNodeLocalDNSAvailable(ctx context.Context, cli client.Client) (bool, error) {
+	ds := &v1.DaemonSet{}
+
+	err := cli.Get(ctx, client.ObjectKey{Namespace: "kube-system", Name: "node-local-dns"}, ds)
+	if err != nil {
+		return false, err
```

Comment:
We should check the error type here and determine if it is 'not found'. If so, we can return `false, nil` - else return `false, err`